soroban-sdk/src/address.rs

```diff
@@ -226,49 +223,55 @@ impl Address {
         self.env.require_auth(self);
     }
 
-    /// Creates an `Address` corresponding to the provided contract identifier.
+    /// Creates an `Address` corresponding to the provided Stellar strkey.
     ///
-    /// Prefer using the `Address` directly as input or output argument. Only
-    /// use this in special cases, for example to get an Address of a freshly
-    /// deployed contract.
+    /// The only supported strkey types are account keys (`G...`) and contract keys (`C...`). Any
+    /// other valid or invalid strkey will cause this to panic.
     ///
-    /// TODO: Replace this function in its pub form with a function that accepts
-    /// a strkey instead. Dependent on <https://github.com/stellar/rs-stellar-strkey/issues/56>.
```

Comment:
Can https://github.com/stellar/rs-stellar-strkey/issues/56 be closed now?

Reply:
https://github.com/stellar/rs-stellar-strkey/issues/56 is still useful independent of this.

Reply:
Ok, let's address it separately then.